server/repository/src/db_diesel/item_ledger.rs

```diff
@@ -147,3 +160,65 @@ fn create_filtered_query(filter: Option<ItemLedgerFilter>) -> ItemLedgerQuery {
 
     query
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        mock::{
+            ledger::{get_test_ledger_datetime, mock_ledger_data},
+            MockData, MockDataInserts,
+        },
+        test_db,
+    };
+
+    use super::*;
+
+    #[actix_rt::test]
+    async fn item_ledger_repository() {
+        // Insert invoice lines for each status to test the view
+        let (items, stock_lines, invoices, invoice_lines) = mock_ledger_data();
+        let (_, storage_connection, _, _) = test_db::setup_all_with_data(
+            "item_ledger_repository",
+            MockDataInserts::all(),
+            MockData {
+                items,
+                stock_lines,
+                invoices,
+                invoice_lines,
+                ..Default::default()
+            },
+        )
+        .await;
+
+        let repo = ItemLedgerRepository::new(&storage_connection);
+        let filter = ItemLedgerFilter::new().item_id(EqualFilter::equal_to("ledger_test_item"));
+
+        let result = repo.query(Pagination::all(), Some(filter)).unwrap();
+
+        // Validate the results based on the mock_ledger_data
+        // PICKED+ outbounds, RECEIVED+ inbounds, VERIFIED adjustments should be included
+
+        assert_eq!(result[0].id, "verified_inventory_adjustment_line");
+        assert_eq!(result[1].id, "picked_outbound_line_stock_line_b");
+        assert_eq!(result[2].id, "picked_outbound_line");
+        assert_eq!(result[3].id, "verified_inbound_line");
+        assert_eq!(result[4].id, "received_inbound_line");
+
+        // There is a ledger entry for another item, check it is not included
+        assert_eq!(result.len(), 5);
+
+        // Check that the results are in the expected order (reverse chronological)
+        assert_eq!(result[0].datetime, get_test_ledger_datetime(5));
+        assert_eq!(result[1].datetime, get_test_ledger_datetime(4));
+        assert_eq!(result[2].datetime, get_test_ledger_datetime(4));
+        assert_eq!(result[3].datetime, get_test_ledger_datetime(3)); // the received time of the verified inbound
+        assert_eq!(result[4].datetime, get_test_ledger_datetime(2));
+
+        // Check the running balance (reverse chronological)
+        assert_eq!(result[0].running_balance, 10.0); // verified inventory addition
+        assert_eq!(result[1].running_balance, 0.0); // picked outbound
```

Comment:
Just checking - we never introduced stock of stock line B before this right? So effectively our running balance here is stockB(-10) + stockA(10) = 0 item units? Then we introduce in the adjustment above some stockB raising it from -10 to 0 (item stock 10).

I realised this when looking at the tests for `ledger.rs`, where the history of stockLine A never reaches 0 despite the *item* history here reaching 0.

Posterity might appreciate if the numbers play fair!

Reply:
Haha I'm still getting confused by the file names I keep having to double check which one is which

![image](https://github.com/user-attachments/assets/ef6f6720-83c6-4698-a027-87dc9c12f548)